tools/tarantool-gdb.py

```diff
@@ -1947,6 +2105,10 @@ def subst_and_eval(indexed_item):
             yield self.child(item_index, item)
 
 pp.add_printer('rlist', '^rlist$', TtListPrinter)
+pp.add_printer('stailq', '^stailq$', TtListPrinter)
+pp.add_printer('stailq_entry', '^stailq_entry$', TtListPrinter)
+if TtStailq.entry_ptr_gdb_type:
+    pp.add_printer('stailq_entry_ptr', '^stailq_entry_ptr$', TtListPrinter)
```

Comment:
The printer doesn't print stailq entries, as it does with rlist:

```
(gdb) p cord_ptr.fiber.storage.txn.stmts
struct stailq<?> of length 2, head=*(stailq*)0x7fd829c0e0b8 = {[0] = 0x7fd829c0e288, [1] = 0x7fd829c0e3a8}
```

Compare with rlist:

```
(gdb) p txm.all_txs
struct rlist<txn::in_all_txs> of length 1, head=*(rlist*)0x55bb487925c8 = {[0] ((txn*)0x7fd829c0e038) = {in_txn_cache = {next = {value = 0x0}}, region = {
      cache = 0x55bb487a1bf0 <main_cord+400>, slabs = {slabs = {prev = 0x7fd829c0e010, next = 0x7fd829c0e010}, stats = {used = 1162, total = 4096}}, on_alloc_cb = 0x0,
      on_truncate_cb = 0x0, cb_arg = 0x0}, acquired_region_used = 0, alloc_stats = {272, 0, 292}, memtx_tx_alloc_stats = 0x7fd829c0e280, id = 391, psn = 0, rv_psn = 0,
    status = TXN_INPROGRESS, isolation = TXN_ISOLATION_BEST_EFFORT, stmts = {first = {value = 0x7fd829c0e288}, last = 0x7fd829c0e3a8}, n_new_rows = 2, n_local_rows = 0,
    n_applier_rows = 0, flags = 0, in_sub_stmt = 0 '\000', sub_stmt_begin = {0x7fd829c0e288, 0x5050505050505050, 0x5050505050505050, 0x5050505050505050}, signature = -1,
    engine = 0x55bb48ff1260, engine_tx = 0x0, fiber = 0x0, start_tm = 1212283.525057215, fiber_on_yield = {link = {prev = 0x7fd829c104f0, next = 0x7fd829c104f0},
      run_links = {prev = 0x7fd829c0e138, next = 0x7fd829c0e138}, run = 0x55bb47fb6061 <txn_on_yield>, data = 0x0, destroy = 0x0}, fiber_on_stop = {link = {
        prev = 0x7fd829c10500, next = 0x7fd829c281d8}, run_links = {prev = 0x7fd829c0e170, next = 0x7fd829c0e170}, run = 0x55bb47fb5f9f <txn_on_stop>, data = 0x0,
      destroy = 0x0}, on_commit = {prev = 0x7fd829c0e198, next = 0x7fd829c0e198}, on_rollback = {prev = 0x7fd829c0e1a8, next = 0x7fd829c0e1a8}, on_wal_write = {
      prev = 0x7fd829c0e1b8, next = 0x7fd829c0e1b8}, savepoints = {prev = 0x7fd829c0e1c8, next = 0x7fd829c0e1c8}, conflict_list = {prev = 0x7fd829c0e1d8,
      next = 0x7fd829c0e1d8}, conflicted_by_list = {prev = 0x7fd829c0e1e8, next = 0x7fd829c0e1e8}, in_read_view_txs = {prev = 0x7fd829c0e1f8, next = 0x7fd829c0e1f8},
    read_set = {prev = 0x7fd829c0e208, next = 0x7fd829c0e208}, point_holes_list = {prev = 0x7fd829c0e218, next = 0x7fd829c0e218}, gap_list = {prev = 0x7fd829c0e228,
      next = 0x7fd829c0e228}, full_scan_list = {prev = 0x7fd829c0e238, next = 0x7fd829c0e238}, in_all_txs = {prev = 0x55bb487925c8 <txm+14856>,
      next = 0x55bb487925c8 <txm+14856>}, is_schema_changed = false, timeout = 3153600000, rollback_timer = 0x0, limbo_entry = 0x0, space_on_replace_triggers_depth = 0}}
```

Also, sometimes listing fails for unknown reason:

```
(gdb) p cord_ptr.fiber.storage.txn.read_set
Warning: failed to identify list (head, entry_info)
Python Exception <class 'AttributeError'> 'NoneType' object has no attribute 'address':
Warning: failed to identify list (head, entry_info)
Python Exception <class 'AttributeError'> 'NoneType' object has no attribute 'address':
{prev = 0x7fd829c0e208, next = 0x7fd829c0e208}
```

How to reproduce:

```Lua
box.cfg{}
box.schema.create_space('test', {if_not_exists = true})
box.space.test:create_index('pk', {if_not_exists = true})
box.begin()
box.space.test:insert{1}
box.space.test:insert{2}
os.execute(string.format('kill -11 %d', require('tarantool').pid()))
```

Reply:
> The printer doesn't print stailq entries, as it does with rlist:
This is because it failed to identify its entry type. In such a case it just display the only information it knows -- the entry link

Reply:
But why did it happen? AFAICS the stailq entry type is present in the list of known stailq types.

Reply:
BTW, I have a problem to get use cases to check with, so thank you for your scenario. I found some problem even if entry info is specified explicitly. I'll let you know as soon as I fix it.

Reply:
> But why did it happen? AFAICS the stailq entry type is present in the list of known stailq types.

Because in this case entry identification mechanism can't "see" that the requested stailq is stored as txn::stmts. It just does see that it is stailq, but doesn't see its container, and thus can't identify it.

A bit longer:
As you may see the "known" types consists of 2 parts (it applies both to `rlist` and `stailq`):
1. symbols (this works if one try to display some list that has corresponding symbol, i.e. global variable, like `rlist engines` or `stailq txn_cache`)
2. various containers fields (this works if the list is a field in container like `main_cord.alive`, but it works only if this container or any outter container has corresponding symbol, otherwise we will not identify that the requested list layout)

Reply:
I don't see any way to workaround the restriction regarding corresponding symbol described above when one uses `print` command, because in this scenario the entry point for custom code is pretty-printer c-tor with `gdb.Value` argument that refers to the list.

But in `tt-list` command we starts with the whole expression and at this point we can figure out container of the list (if any) even if there is no corresponding symbol. And this is exactly what I'm working on at the moment